kahuna/public/js/upload/jobs/upload-jobs.js

```diff
@@ -43,93 +45,124 @@ jobs.controller('UploadJobsCtrl', [
-                // archived, etc), refresh the copy we hold
-                $rootScope.$on('images-updated', (e, updatedImages) => {
-                  const maybeUpdateImage = updatedImages.find(updatedImage => updatedImage.data.id === image.data.id);
-                    if (maybeUpdateImage !== undefined) {
-                        jobItem.image = maybeUpdateImage;
-                    }
-                });
-
-                // we use the filename of the image if the description is missing
-                if (!jobItem.image.data.metadata.description) {
-                    const newDescription = jobItem.name
-                        .substr(0, jobItem.name.lastIndexOf('.'))
-                        .replace(/_/g, ' ');
-
-                    editsService.updateMetadataField(jobItem.image, 'description', newDescription);
-                }
-
-                if (presetLabels.length > 0) {
-                    labelService.add(image, presetLabels);
-                }
-
-                $rootScope.$emit(
-                  'track:event',
-                  eventName,
-                  null,
-                  'Success',
-                  null,
-                  { 'Labels' : presetLabels.length}
-                );
-            }, error => {
-                jobItem.status = 'upload error';
-                jobItem.error = error.message;
-
-                $rootScope.$emit(
-                  'track:event',
-                  eventName,
-                  null,
-                  'Failure',
-                  null,
-                  { 'Failed on': 'index'}
-                );
-            });
+           jobItem.status = 'indexing';
+           jobItem.resource = resource;
+
+           // TODO: grouped polling for all resources we're interested in?
+           const imageUploadStatusResource = apiPoll(() => getUploadStatus(resource));
+           imageUploadStatusResource.then(status => {
+             if (status.data.status === "FAILED"){
+                 const quarantineNotificationDiv = document.getElementById(jobItem.name);
+                 quarantineNotificationDiv.innerHTML = `upload error (${status.data.errorMessage})`;
+                 quarantineNotificationDiv.classList.add("status--invalid");
+                 ctrl.remaining -= 1;
+             } else if (status.data.status === "COMPLETED"){
+                  const findImage = () => status.get();
+                  const imageResource = apiPoll(findImage);
+                  imageResource.then(image => {
+                      jobItem.status = 'uploaded';
+                      jobItem.image = image;
+                      jobItem.thumbnail = image.data.thumbnail;
+
+                      ctrl.remaining -= 1;
+
+                      imageService(image).states.canDelete.then(deletable => {
+                           jobItem.canBeDeleted = deletable;
+                      });
+
+                      // TODO: we shouldn't have to do this ;_;
```

Comment:
```suggestion
```
This TODO is also probably worth removing. As sad as people were that `$rootScope.on` was used for state updates, it's used everywhere, and seems to be the only way that works.